crates/starknet-os/src/hints/execution.rs

```diff
@@ -1119,6 +1121,21 @@ pub fn check_new_deploy_response(
     _constants: &HashMap<String, Felt252>,
 ) -> Result<(), HintError> {
     let response_ptr = get_ptr_from_var_name(vars::ids::RESPONSE, vm, ids_data, ap_tracking)?;
```

Comment:
@notlesh cant we use the "need_retdata_hack" here?

Reply:
I don't think so. The `need_retdata_hack` is part of the syscall struct, which has disappeared by this point. It's possible that we could write something to the vm to use later, but I think that's a bad idea.